### ISSUE TITLE
Any plan to upgrade to Zuul 2

### ISSUE THREAD

**Issue by surajrrao** [spring-cloud/spring-cloud-netflix#1498]
Any plan to upgrade to Zuul 2

**Comment by spencergibb** [comment]
Zuul 2 has not been released. It will not be a simple upgrade when it is. We'll consider it when it is released.

**Comment by robertofpaf** [comment]
Zuul 2 has now been released (based on https://github.com/Netflix/zuul/issues/298), so are there any plans to support it in the foreseeable future?

**Comment by spencergibb** [comment]
We have no plans to integrate zuul 2

**Comment by sharmasourabh** [comment]
Hi @spencergibb 
Spring Cloud is not going to support future versions of all Netflix libraries like Eureka etc or only Zuul? Is there any publish plan or road map that gives us this information?
Thanks.

**Comment by ryanjbaxter** [comment]
@sharmasourabh this is only a statement regarding Zuul, it does not effect any other Spring Cloud components.

**Comment by sharmasourabh** [comment]
Thanks @spencergibb for confirming.

**Comment by robertofpaf** [comment]
That would be good to know. I understand Zuul 2 is completely different and it won't be trivial to integrate it with the other Spring Cloud components, but Zuul 1's lack of support for websockets, server sent events, HTTP/2 makes it less and less suitable for modern applications.

**Comment by juergenzimmermann** [comment]
IMHO the new Spring Cloud Gateway looks promising.

**Comment by spencergibb** [comment]
http://slides.com/spencer/spring-cloud-gateway

**Comment by sharmasourabh** [comment]
I hope, Spring Cloud Gateway performance is/would improved down the line. 
As shown in this performance comparison: https://engineering.opsgenie.com/comparing-api-gateway-performances-nginx-vs-zuul-vs-spring-cloud-gateway-vs-linkerd-b2cc59c65369

**Comment by spencergibb** [comment]
That is outdated

**Comment by ghoddg** [comment]
but it seems we have some open issues with Spring Cloud Gateway e.g.

java.lang.ClassCastException: org.springframework.core.io.buffer.DefaultDataBufferFactory cannot be cast to org.springframework.core.io.buffer.NettyDataBufferFactory
        at org.springframework.cloud.gateway.filter.NettyWriteResponseFilter.lambda$filter$0(NettyWriteResponseFilter.java:71) ~[spring-cloud-gateway-core-2.0.0.BUILD-SNAPSHOT.jar:2.0.0.BUILD-SNAPSHOT]

**Comment by spencergibb** [comment]
Not appropriate to comment on the gateway here

**Comment by gihad** [comment]
It would be great to get some reasonable responses to the questions in this issue:

Q: "Any chance to get a motivation why Zuul 2 Spring Cloud isn't happening? Is it because of the very different non-blocking i/o?"


Q: " hope, Spring Cloud Gateway performance is/would improved down the line.
As shown in this performance comparison: https://engineering.opsgenie.com/comparing-api-gateway-performances-nginx-vs-zuul-vs-spring-cloud-gateway-vs-linkerd-b2cc59c65369"
A: "That is outdated"

Is there an up to date benchmark available?

**Comment by gihad** [comment]
Thinking a bit more about it, since I saw the early version of Spring Webflux and that Spring was creating their own non-blocking async programming model I started doubting that Zuul2 was ever gonna be a part of future Spring versions, it makes sense.

I've also been waiting for Spring Webflux to become performant to use it in my apps instead of Vert.x for async non-blocking. The benchmarks in general for Spring Webflux shows low performance compared to other modern concurrency models for the JVM.

Since spring-cloud-gateway uses Spring Webflux as the underlining tech it seems reasonable that it doesn't have the greatest performance yet. So now we just need to wait until this catches up.

At the end of the day it will be nice to have a "Gateway" that is even more integrated into the spring ecosystems and leverages existing Spring components.
